packages/block-editor/src/components/button-block-appender/index.native.js

```diff
@@ -39,7 +40,7 @@ function ButtonBlockAppender( {
 				rootClientId={ rootClientId }
 				renderToggle={ ( { onToggle, disabled, isOpen } ) => (
 					<Button
-						onClick={ onToggle }
+						onClick={ customOnAdd || onToggle }
```

Comment:
I would try to do the same as here https://github.com/WordPress/gutenberg/pull/19013/files#r400259304

Reply:
In this case `onToggle` opens the BottomSheet and `customOnAdd` automatically adds the new block. I'm not sure if we want to mix it together.